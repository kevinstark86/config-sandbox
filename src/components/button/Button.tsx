// eslint-disable-next-line import/no-extraneous-dependencies
import styled from 'styled-components';
import {BtnProps} from './types';

const Wrapper = styled.button<BtnProps>`
  padding: 10px 40px;
  border: none;
  border-radius: 5px;
  background-color: ${props => (props.variant === 'primary' ? 'blue' : 'red')};
  color: ivory;
`;

function Button({variant, disabled, children}: BtnProps): JSX.Element {
  return (
    <Wrapper variant={variant} disabled={disabled} type="button">
      {children}
    </Wrapper>
  );
}

export default Button;
