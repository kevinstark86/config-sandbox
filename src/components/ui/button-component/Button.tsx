// eslint-disable-next-line import/no-extraneous-dependencies
import styled, {css} from 'styled-components';
import {BtnProps} from './types';

// Component Styling
const Wrapper = styled.button<BtnProps>`
  padding: 10px 40px;
  border-radius: 5px;
  cursor: pointer;
  ${props =>
    props.variant === 'primary' &&
    css`
      background-color: blue;
      color: ivory;
      border: none;
    `}
  ${props =>
    props.variant === 'secondary' &&
    css`
      border: 2px solid blue;
      color: blue;
      background-color: transparent;
    `}
    ${props =>
    props.variant === 'warning' &&
    css`
      border: none;
      color: ivory;
      background-color: red;
    `}
    ${props =>
    props.disabled === true &&
    css`
      background-color: grey;
      border: none;
      color: ivory:
    `}
`;

function Button({variant, disabled, children}: BtnProps): JSX.Element {
  return (
    <Wrapper variant={variant} disabled={disabled} type="button">
      {children}
    </Wrapper>
  );
}

export default Button;
