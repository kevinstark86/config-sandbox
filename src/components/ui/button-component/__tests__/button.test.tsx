/* eslint-disable import/no-extraneous-dependencies */
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import Button from '../Button';

describe('render variant styles', () => {
  test('primary variant', () => {
    const tree = renderer
      .create(
        <Button variant="primary" disabled={false}>
          Click Me
        </Button>
      )
      .toJSON();
    expect(tree).toHaveStyleRule('background-color', 'blue');
    expect(tree).toHaveStyleRule('color', 'ivory');
    expect(tree).toHaveStyleRule('cursor', 'pointer');
  });
  test('secondary variant', () => {
    const tree = renderer
      .create(
        <Button variant="secondary" disabled={false}>
          Click Me
        </Button>
      )
      .toJSON();
    expect(tree).toHaveStyleRule('background-color', 'transparent');
    expect(tree).toHaveStyleRule('color', 'blue');
    expect(tree).toHaveStyleRule('border', '2px solid blue');
    expect(tree).toHaveStyleRule('cursor', 'pointer');
  });
  test('warning variant', () => {
    const tree = renderer
      .create(
        <Button variant="warning" disabled={false}>
          Click Me
        </Button>
      )
      .toJSON();
    expect(tree).toHaveStyleRule('background-color', 'red');
    expect(tree).toHaveStyleRule('color', 'ivory');
    expect(tree).toHaveStyleRule('cursor', 'pointer');
  });
});

describe('button functionality', () => {
  test('button renders children', () => {
    render(
      <Button variant="primary" disabled={false}>
        Click Me
      </Button>
    );
    const button = screen.getByRole('button', {name: /click me/i});
    expect(button).toBeInTheDocument();
    expect(button).toHaveTextContent('Click Me');
  });

  test('button is disabled', () => {
    render(
      <Button variant="primary" disabled>
        Click Me
      </Button>
    );
    const button = screen.getByRole('button', {name: /click me/i});
    expect(button).toBeInTheDocument();
    expect(button).toBeDisabled();
  });
});
