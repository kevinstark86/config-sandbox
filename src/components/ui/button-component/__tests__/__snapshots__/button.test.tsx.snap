// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render test 1`] = `
.c0 {
  padding: 10px 40px;
  border-radius: 5px;
  background-color: blue;
  color: ivory;
  border: none;
}

.c0:hover {
  cursor: pointer;
}

<button
  className="c0"
  disabled={false}
  type="button"
>
  Click Me
</button>
`;
